libraries/opensk/src/api/crypto/sha256.rs

```diff
@@ -16,11 +16,23 @@ use super::HASH_SIZE;
 
 /// Hashes data using SHA256.
 pub trait Sha256: Sized {
-    /// Computes the hash of a given message directly.
+    /// Computes the hash of a given message as an array.
+    ///
+    /// This function does not let you control the memory allocation. It should therefore not be
+    /// used for secrets that need zeroization.
     fn digest(data: &[u8]) -> [u8; HASH_SIZE] {
         let mut hasher = Self::new();
         hasher.update(data);
-        hasher.finalize()
+        let mut output = [0; HASH_SIZE];
+        hasher.finalize(&mut output);
+        output
+    }
```

Comment:
Same here, the non-`_mut` version should be implemented using the `_mut` version.